### DISCUSSION TITLE
LSP keeps on getting restarted on every edit / file save

### DISCUSSION THREAD

**Discussion by sourabhxyz** [LazyVim/LazyVim#883] [Q&A]
Hi, when working on Haskell project, whenever I make a small change (even a change in comment), or save the file, I begin seeing my LSP redoing the work, i.e., I see "Processing hls" (hls - Haskell Language Server) in bottom right, as in image.

![issue](https://github.com/LazyVim/LazyVim/assets/29835247/348335dc-512e-4b39-884b-bac9a770c403)

In VSCode, I don't see such an issue and my HLS always have files processed, i.e., immediately after an edit, I can see it's results and there is no lag.

In Vim, before moving to LazyVim, I didn't notice this issue.

Besides defaults, I have added the file with following contents to `plugins` folder:

```lua
return {
  -- add hls to lspconfig
  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        -- hls will be automatically installed with mason and loaded with lspconfig
        hls = {
          settings = {
            haskell = {
              formattingProvider = "stylish-haskell",
            },
          },
        },
      },
    },
  },
}
```

Any help is deeply appreciated :pray:!

**Comment by folke** [comment]
This has nothing to do with LazyVim. Just your lsp

**Reply by sourabhxyz** [reply]
Thanks for prompt response @folke! But are you really sure about it? Isn't this LSP editor agnostic? I rechecked that this LSP immediately gives result in VSCode but not here.

But I guess I am missing something.

**Reply by folke** [reply]
What I mean, that this is not a LazyVim issue. LazyVim just uses lspconfig.
